Research/ResearchUI/iOS/Step View Controllers/RSDScrollingOverviewStepViewController.swift

```diff
@@ -88,6 +108,22 @@ open class RSDScrollingOverviewStepViewController: RSDOverviewStepViewController
                 iconImages[idx].image = iconInfo.icon?.embeddedImage()
                 iconTitles[idx].text = iconInfo.title
             }
+            
+            // When there are only 2 icons, we need to make adjustments to center them
+            if (icons.count == 2) {
+                let removeIdx = 2
+                
+                // Adjust margin factor to give smaller screens more room
+                let cellWidth = (iconImages[removeIdx].superview?.superview?.frame.size.width ?? 0) / CGFloat(iconImages.count)
+                let marginFactor: CGFloat = (cellWidth < 125) ? 3.0 : 2.0
+                
+                // First, Adjust the leading/trailing spacing
+                iconImagesLeadingConstraint.constant = cellWidth / marginFactor
+                iconImagesTrailingConstraint.constant = cellWidth / marginFactor
+                
+                // Then, remove the third icon from the stack view
+                iconImages[removeIdx].superview?.removeFromSuperview()
+            }
```

Comment:
So... yeah. If you look higher up in this code file there is a TODO about refactoring to use a UICollectionView. This will handle the case for 1, 2, 3 icons but... what about 4? Please add or move the TODO comment because this is still just another hack and should be explained. ;)